rustsat/src/types.rs

```diff
@@ -981,4 +1041,56 @@ mod tests {
     fn ternary_val_size() {
         assert_eq!(size_of::<TernaryVal>(), 1);
     }
+
+    #[test]
+    fn parse_vline() {
+        let vline = "v 1 -2 4 -5 6 0";
+        let ground_truth = Assignment::from(vec![
+            TernaryVal::True,
+            TernaryVal::False,
+            TernaryVal::DontCare,
+            TernaryVal::True,
+            TernaryVal::False,
+            TernaryVal::True,
+        ]);
+        assert_eq!(Assignment::from_vline(vline).unwrap(), ground_truth);
+        assert_eq!(
+            {
+                let mut asign = Assignment::default();
+                asign.extend_from_vline(vline).unwrap();
+                asign
+            },
+            ground_truth
+        );
+    }
+
+    #[test]
+    #[should_panic]
+    fn vline_invalid_lit_from() {
+        let vline = "v 1 -2 4 foo -5 bar 6 0";
+        Assignment::from_vline(vline).unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn vline_invalid_lit_extend() {
+        let vline = "v 1 -2 4 foo -5 bar 6 0";
+        let mut assign = Assignment::default();
+        assign.extend_from_vline(vline).unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn vline_invalid_tag_from() {
+        let vline = "b 1 -2 4 -5 6 0";
+        Assignment::from_vline(vline).unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn vline_invalid_tag_extend() {
+        let vline = "b 1 -2 4 -5 6 0";
+        let mut assign = Assignment::default();
+        assign.extend_from_vline(vline).unwrap();
+    }
```

Comment:
Once you have the `InvalidVLine` error type implemented, feel free to change these to check for the exact error with `downcast` as I mentioned above.